core/builtins/native/kotlin/Nothing.kt

```diff
@@ -17,6 +17,7 @@
 package kotlin
 
 /**
- * Nothing has no instances
+ * Nothing has no instances. You can use Nothing as the return type of a function to indicate
+ * that it never returns (always throws an exception).
```

Comment:
There are more use-cases for Nothing, e.g. Iterable<Nothing> is ok to exist, but it will never enter "for" loop, etc. Needs link to lang docs.